ansible_rulebook/engine.py

```diff
@@ -350,6 +344,17 @@ async def run_ruleset(
             continue
         results = []
         try:
+            # Asset added by other ruleset, not related to current event
+            while not ruleset_queue_plan.plan.queue.empty():
+                item = cast(
+                    ActionContext, await ruleset_queue_plan.plan.queue.get()
+                )
+                result = await call_action(*item, event_log=event_log)
+
+            # create a new plan queue for current event so that results
+            # can be concanated
+            ruleset_queue_plan.plan.queue = asyncio.Queue()
```

Comment:
@bzwei I am not sure if this is in the correct place. If we have events flowing thru the system wether they match a rule or not we create these Queue objects for every event. So for a simple range source plugin that generates 10 events and you have a condition that says event.i > 8, this code will unconditionally create 10 queues when only 2 might be needed.

Reply:
The queue is needed by [lang.post(name, data)](https://github.com/ansible/ansible-rulebook/pull/202/files#diff-33ca52e665e828cd48de6eec0723f8b5e74fbcc7ecda7df2d392171ae1987365L354). It may or may not added an item to the queue, but we don't know before hand.

Reply:
@bzwei This lang.post code doesn't use any queues, its targeting a running ruleset by name and sending that data directly to the Drools rule engine.
It hits this post code in the drools_jpy and gets forwarded to Drools which processes all the rules.

https://github.com/mkanoor/drools_jpy/blob/c0936793d06a7231b8a64bd7c4929a009b3fd934/src/drools/ruleset.py#L151


This picture might help clarify things

<img width="772" alt="Screen Shot 2022-10-14 at 7 28 53 PM" src="https://user-images.githubusercontent.com/6452699/195957412-22f42d12-76ca-4dd7-8390-0a2b76076d07.png">

All Post Event/Assert Fact/Set Event don't go thru queues they are directly sent to the running ruleset in both Drools and durable rules.